source/Ship.cpp

```diff
@@ -4739,6 +4812,23 @@ void Ship::CreateSparks(vector<Visual> &visuals, const Effect *effect, double am
 
 
 
+
+// Move the given ship into one of the bays (if possible) and generate any visuals.
+bool Ship::Carry(const shared_ptr<Ship> &ship, vector<Visual> &visuals)
+{
+	bool success = Carry(ship);
+
+	// Apply any retrieve effects. If the ship was carried succesfully then
+	// it was assigned a bay with the effects to generate.
+	if(success && !ship->reservedBay->retrieveEffects.empty())
+		for(const Effect *effect : ship->reservedBay->retrieveEffects)
```

Comment:
No difference in behavior, I imagine.
```suggestion
	if(success)
		for(const Effect *effect : ship->reservedBay->retrieveEffects)
```